geopandas/io/file.py

```diff
@@ -2,29 +2,58 @@
 
 import fiona
 import numpy as np
+import six
 
 from geopandas import GeoDataFrame
 
+# Adapted from pandas.io.common
+if six.PY3:
+    from urllib.request import urlopen as _urlopen
+    from urllib.parse import urlparse as parse_url
+    from urllib.parse import uses_relative, uses_netloc, uses_params
+else:
+    from urllib2 import urlopen as _urlopen
+    from urlparse import urlparse as parse_url
+    from urlparse import uses_relative, uses_netloc, uses_params
+
+_VALID_URLS = set(uses_relative + uses_netloc + uses_params)
+_VALID_URLS.discard('')
+
+
+def _is_url(url):
+    """Check to see if *url* has a valid protocol."""
+    try:
+        return parse_url(url).scheme in _VALID_URLS
+    except:
+        return False
+
 
 def read_file(filename, **kwargs):
     """
-    Returns a GeoDataFrame from a file.
+    Returns a GeoDataFrame from a file or URL.
 
-    *filename* is either the absolute or relative path to the file to be
-    opened and *kwargs* are keyword args to be passed to the `open` method
-    in the fiona library when opening the file. For more information on
-    possible keywords, type: ``import fiona; help(fiona.open)``
+    *filename* is either the absolute or relative path to the file or URL to
+    be opened and *kwargs* are keyword args to be passed to the `open`
+    or (`BytesCollection`) method in the fiona library when opening the file.
+    For more information on possible keywords, type:
+    ``import fiona; help(fiona.open)``
```

Comment:
I think it would be valuable to expand the docs a bit here now with this added functionality. In the `from_file` method on a geodataframe, which wraps `read_file`, there's a basic example. Something like that would be useful here. Additionally, putting it in `numpy` style to help with effort to make formatting of docs consistent.